lib/qa/authorities/persons.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+require 'identity'
+
+module Qa
+  module Authorities
+    class Persons < Base
+      attr_reader :term
+
+      def search(term)
+        @term = term
+
+        persons.map do |person|
+          formatted_response(person)
+        end
+      end
+
+      private
+
+        # @return Array<Creator, Identity::Person>
+        # @note if the same person is in both sources, prefer the Creator over the Identity::Person
+        def persons
```

Comment:
Even though this is totally meaningless on such a small scale, I have a hard time abiding the O(n^2) runtime here. What about this:

```ruby
creator_ids = Set.new( creators.map(&:psu_id) )
(creators + identities).reject do |person|
  person.is_a?(Identity::Person) && creator_ids.include?(person.user_id)
end
```

Reply:
@rschenk If I'm understanding this right, this doesn't make the problem go away, but mitigates it because `Set` is faster?

Granted, it will be a small set, but we could ensure that by limiting the number of each identity and creator returned.

Another idea I had was to create a hybrid class that encapsulates both the data types and rejects duplicates as they are added. This would also make the `formatted_person` and `formatted_creator` hashes go away.

Reply:
@rschenk cool. I benchmarked the two variations and ran the unit test:
```
>>>>>>>>>>>>>>>>>>> With Set >>>>>>>>>>>>>>>>>>>>
I, [2020-04-07T10:51:52.631979 #64522]  INFO -- :   0.000745   0.000044   0.000789 (  0.001045)

I, [2020-04-07T10:51:52.656366 #64522]  INFO -- :   0.000497   0.000025   0.000522 (  0.000634)

I, [2020-04-07T10:51:52.693092 #64522]  INFO -- :   0.000484   0.000020   0.000504 (  0.000623)

I, [2020-04-07T10:51:52.707997 #64522]  INFO -- :   0.000473   0.000028   0.000501 (  0.000662)

I, [2020-04-07T10:51:52.722029 #64522]  INFO -- :   0.000503   0.000033   0.000536 (  0.000708)

I, [2020-04-07T10:51:52.834403 #64522]  INFO -- :   0.000503   0.000021   0.000524 (  0.000700)

>>>>>>>>>>>>>>>>>>> With Map >>>>>>>>>>>>>>>>>>>>
I, [2020-04-07T10:53:05.640331 #65263]  INFO -- :   0.013883   0.002076   0.015959 (  0.016295)

I, [2020-04-07T10:53:05.659549 #65263]  INFO -- :   0.000407   0.000065   0.000472 (  0.000569)

I, [2020-04-07T10:53:05.674009 #65263]  INFO -- :   0.000397   0.000013   0.000410 (  0.000504)

I, [2020-04-07T10:53:05.686974 #65263]  INFO -- :   0.000461   0.000026   0.000487 (  0.000621)

I, [2020-04-07T10:53:05.701285 #65263]  INFO -- :   0.000506   0.000054   0.000560 (  0.000737)

I, [2020-04-07T10:53:05.761180 #65263]  INFO -- :   0.000396   0.000017   0.000413 (  0.000502)
```
The first line in the log of each section is the one to look at.